portfolio/src/main/java/com/google/sps/data/TimeRange.java

```diff
@@ -0,0 +1,79 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+/**
+ * Class representing a span of time, enforcing properties (e.g. start comes before end) and
+ * providing methods to make ranges easier to work with (e.g. {@code overlaps}).
+ */
+public final class TimeRange {
+  private final long start;
+  private final long duration;
+
+  private TimeRange(long start, long duration) {
+    this.start = start;
+    this.duration = duration;
+  }
+
+  /** Returns the start of the range in minutes. */
+  public long start() {
+    return start;
+  }
+
+  /** Returns the number of minutes between the start and end. */
+  public long duration() {
+    return duration;
+  }
+
+  /** Returns the end of the range. This ending value is the closing exclusive bound. */
+  public long end() {
+    return start + duration;
+  }
+
+  @Override
+  public boolean equals(Object other) {
+    return other instanceof TimeRange && equals(this, (TimeRange) other);
+  }
+
+  @Override
```

Comment:
I feel like we might end up not needing this hashCode function, so maybe it's okay to delete?

Reply:
Done.